packages/components/bucket-policy/bucket-policy-body.tsx

```diff
@@ -0,0 +1,442 @@
+import { LaunchModal } from '../../utils/modals/modalLauncher';
+import { getName, getUID, getNamespace } from '../../utils/selectors/k8s';
+import {
+  ReplicationOBC,
+  BucketPolicyState,
+  BucketPolicyAction,
+  BucketPolicyActionType,
+} from './state';
+
+const onChangeMulti: onChangeMultiSelectProps =
+  (data, dispatch) => (selection) => {
+    const alreadyIncluded = data.includes(selection);
+    const payload = alreadyIncluded
+      ? data.filter((resource) => resource !== selection)
+      : [...data, selection];
+    dispatch({
+      type: BucketPolicyActionType.SET_READ_MULTI,
+      payload,
+    });
+  };
+
+/*const onChangeReplication: onChangeReplicationSelectProps =
+  (replicationObj, dispatch) => (resource) => {
+    const alreadyIncluded = replicationObj.some((obj) => obj.obcName === resource?.metadata?.name);
+    const payload: ReplicationOBC[] = alreadyIncluded
+      ? replicationObj.filter((obj) => obj.obcName !== resource?.metadata?.name)
+      : [...replicationObj, {
+        obcName: resource?.metadata?.name,
+        targetBucket: resource?.spec?.bucketName,
+      }];
+    dispatch({
+      type: BucketPolicyActionType.SET_REPLICATION_OBC,
+      payload,
+    })
+  };*/
+
+const DataResourceDropdown: React.FC<DataResourceDropdownProps> = ({
+  data,
+  onChange,
+  launchModal,
+  disableAllExcept,
+  variant = SelectVariant.single,
+  ...props
+}) => {
+  const { t } = useTranslation();
+  const [isOpen, setOpen] = React.useState(false);
+
+  const dropdownOptions: JSX.Element[] = React.useMemo(
+    () =>
+      data.reduce(
+        (acc, dataResource) => {
+          const name = getName(dataResource);
+          const isDisabled =
+            (!_.isEmpty(disableAllExcept) &&
+              !disableAllExcept?.some((itemName) => itemName === name)) ||
+            dataResource?.status?.phase !== 'Ready';
+          acc.push(
+            <SelectOption
+              key={getUID(dataResource)}
+              value={name}
+              isDisabled={isDisabled}
+              data-test={`${name}-dropdown-item`}
+            />
+          );
+          return acc;
+        },
+        [
+          <Button
+            key="data-source-add-modal"
+            className="create-data-source__modal-button"
+            variant="plain"
+            onClick={() => {
+              setOpen(false);
+              launchModal();
+            }}
+          >
+            {t('Add new data source')}
+          </Button>,
+          <Divider component="li" key={'data-source-divider'} />,
+        ]
+      ),
+    [t, data, disableAllExcept, launchModal]
+  );
+
+  return (
+    <Select
+      {...props}
+      className={classNames(props?.className)}
+      variant={variant}
+      aria-label={t('Select data source')}
+      onToggle={setOpen}
+      onSelect={(e, selection) => {
+        onChange(selection as string);
+        variant !== SelectVariant.checkbox && setOpen(false);
```

Comment:
Is this check for the first modal button ? The button onClick should do the job right ?

Reply:
not the first... basically for all the other elements after the first one...